### PR TITLE
Build: enable building for Debian 11

### PR DESCRIPTION
**Pull request by pstorz**
#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

##### General

- [x] PR name is meaningful
- [x] Purpose of the PR is understood
- [x] Separate commit for this PR in the CHANGELOG.md, PR number referenced is same
- [x] Commit descriptions are understandable and well formatted

##### Source code quality

- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [ ] Required documentation changes are present and part of the PR
- [x] `bareos-check-sources --since-merge` does not report any problems
- [x] `git status` should not report modifications in the source tree after building and testing

##### Tests

- [ ] ~Decision taken that a system- or unittest is required (if not, then remove this paragraph)~
- [ ] ~The decision towards a systemtest is reasonable compared to a unittest~
- [ ] ~Testname matches exactly what is being tested~
- [ ] ~Output of the test leads quickly to the origin of the fault~

### CODE REVIEW

File: debian/control.in [hunk 1]
```diff
@@ -45,10 +45,9 @@ Build-Depends: acl-dev,
...
  pkg-config,
  po-debconf (>= 0.8.2),
- python-dev,
+# python-dev,
  zlib1g-dev,
  systemd,
- dh-systemd,
  librados-dev <buster> <stretch> <xenial> <bionic>,
  libradosstriper-dev <buster> <stretch> <xenial> <bionic>,
  libcephfs-dev <buster> <stretch> <xenial> <bionic>,
```
Comment (arogge) [inline debian/control.in:53]: these conditionals will need to have `<bullseye>` added to work as expected for us

File: debian/control.in [hunk 2]
```diff
@@ -45,10 +45,9 @@ Build-Depends: acl-dev,
  ncurses-dev,
  pkg-config,
  po-debconf (>= 0.8.2),
- python-dev,
+# python-dev,
  zlib1g-dev,
  systemd,
- dh-systemd,
```
Comment (arogge) [inline debian/control.in:51]: maybe we should tag it with the versions where we want it like this:
```
dh-systemd <buster> <stretch> <xenial> <bionic>,
```
This way at least our own builds will require it for these older releases.

**Review by arogge — CHANGES_REQUESTED** [review]
We seem to have to remove dh-python as it conflicts with python-is-python2 and python-is-python3. However, I believe the real fix for this is not to use a versionless-python anymore.
We can probably require python2-dev and python3-dev for Bareos itself and dh-python for python-bareos resulting in clean dependencies.
AFAICT the only uses of a plain `python` are in `python-bareos/setup.py` and `core/cmake/get_python_compile_settings.py` - which seems to be easily fixable.

File: debian/control.in [hunk 3]
```diff
@@ -45,10 +45,9 @@ Build-Depends: acl-dev,
  ncurses-dev,
  pkg-config,
  po-debconf (>= 0.8.2),
- python-dev,
+# python-dev,
```
[the remark this answers is not in the document]
Comment (pstorz) [reply debian/control.in:48]: The bug that required thjis change is here:
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=972726

File: debian/control.in (hunk 2 above)
Comment (pstorz) [reply debian/control.in:51]: Good point

**Comment by aba-rechsteiner** [comment]
Are there any plans by when the packages for Debian 11 will be ready?

**Review by arogge — CHANGES_REQUESTED** [review]
You need to add a record to the CHANGELOG, but otherwise it looks fine!

**Pull request merged**